src/ui.tsx

```diff
@@ -1084,6 +1117,11 @@ function Root() {
     };
   });
 
+  // Each time new line is started, save the start time
+  const currentLineStartedTime = useMemo(() => {
```

Comment:
nit: rename this to `currentMotionStartedTime`

the `progress` updates not just when the pen is drawing a line, but also when the pen is raising/lowering, and when the pen is moving in the air. The semantics are correct as written, but the name could be clearer. We use 'motion' elsewhere in the codebase to represent any movement of the machine, so I think that's the appropriate term here too :)